client/genesis-builder/src/lib.rs

```diff
@@ -0,0 +1,239 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::traits::BuildGenesisConfig;
```

Comment:
@bkchr 
It seems that this helper cannot be placed in `client` folder, because it depends on `frame_support` crate ([test_dependency complains](https://gitlab.parity.io/parity/mirrors/substrate/-/jobs/3042323)). What would be better place to put it in?

(The intention here is to provide some common logic in single helper that can be easily used in runtime implementation e.g.:
https://github.com/paritytech/substrate/blob/3e130fb25053f9d237d443fa4a4281098048e1d7/test-utils/runtime/src/lib.rs#L718-L732
)

Reply:
We could move `BuildGenesisConfig` into `sp-runtime`, what would also allow for blanket implementation for pallets' GenesisConfig (instead of generating it in macro).

Reply:
I don't get the need for this crate? The `GenesisBuilderHelper` is a struct that is FRAME specific and you put it into a client crate? You should add this to `frame-support`.